handler/handler_test.go

```diff
@@ -76,6 +75,16 @@ func (m *MockMetricStore) Ready() error {
 	return nil
 }
 
+func ctxWithParams(params map[string]string, mainReq *http.Request) context.Context {
```

Comment:
@beorn7 Probably changing this function can solve problems occurring in test cases